hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -691,7 +691,10 @@ class ArrayStructExpression(ArrayExpression):
     """
 
     def __getattr__(self, item):
-        return ArrayStructExpression.__getitem__(self, item)
+        try:
+            return ArrayStructExpression.__getitem__(self, item)
+        except KeyError as e:
+            raise AttributeError(str(e)) from e
```

Comment:
should be `AttributeError(item) from e` I think?

Reply:
`AttributeError(f"'{self.__class__.__name__}' object has no attribute '{item}'")` would be consistent with the default behavior of objects. But the message from the KeyError is formatted by `hail.utils.misc. get_nice_field_error`.